p_sets/p_set_1.rb

```diff
@@ -0,0 +1,91 @@
+# http://cdn.cs50.net/2016/x/psets/1/pset1/pset1.html
+# SMART WATER
+BOTTLE_VOLUME = 16
+WATER_FLOW_RATE = 192
+
+def bottle_reporter
+  puts "That's #{bottle_calculator ask_minutes} bottles of water"
+end
+
+def bottle_calculator(minutes)
+  minutes * WATER_FLOW_RATE / BOTTLE_VOLUME
+end
+
+def ask_minutes
+  puts "How many mins do you shower for?"
+  minutes = gets.chomp.to_i
+end
+
+# ITSA MARIO
+SPACE =      " "
+BRICKS =     "#"
+APEX_WIDTH = 2
+MIN_HEIGHT = 0
+MAX_HEIGHT = 23
+
+def ask_height
+  while true
+    puts "height?"
+    height = gets.chomp.to_i
+    break if valid_input?(height)
+  end
+  height
+end
+
+def valid_input?(height)
+  height <= MAX_HEIGHT && height > MIN_HEIGHT
```

Comment:
Although this section was about comparison operators, I think it's useful to consider Ruby's Range objects here:

```
VALID_HEIGHTS = MIN_HEIGHT..MAX_HEIGHT
# …
VALID_HEIGHTS.include? height
```